### ISSUE TITLE
Nano: we should support a flag that indicates the model does not have batchsize dim

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#6391]
## background
This is a common case when it comes to Graph Neural Network like GCN. A typical task for GCN is node classification, where GCN(and many other GNN) takes 2 inputs, one for node features with shape (node_num, feature_dim) and the other for adj matrix (potentially sparse) with shape (node_num, node_num)

## why it breaks our current implementation?
Many implementation takes the first dim as batchsize for granted. This inclues
1. INC related functions require a dataloader with batchsize dim
2. our metric implementation takes a dataloader with batchsize dim

## how to solve
we may provide 

1. a repeat dataset for users who only have 1 sample of data to inference
```python
class RepeatDataset(Dataset):
    def __init__(self, data, target, num=100):
        self.data = data
        self.target = target
        self.length = num

    def __len__(self):
        return self.length

    def __getitem__(self, idx):
        return self.data
```

2. a `pre_hook` function implemented similar to
```python
def pre_hook(module, inp):
    return tuple(map(lambda x:x[0], inp))
```
to eliminate the batchsize dim added by `RepeatDataset`
